lib/spack/spack/util/web.py

```diff
@@ -222,10 +224,31 @@ def handle_starttag(self, tag, attrs):
                     self.base_url = val
 
 
+def fetch_from_ssh_url(url, dest):
+    if isinstance(url, str):
+        url = urllib.parse.urlparse(url)
+
+    scp = spack.util.executable.which("scp", required=True)
+    scp_args = [] if tty.is_debug() else ["-q", "-o", "LogLevel=QUIET"]
+    scp(*scp_args, f"{url.netloc}:{url.path}", dest, fail_on_error=True)
+
+
+def read_from_ssh_url(url):
+    with tempfile.NamedTemporaryFile("rb") as tmp:
```

Comment:
Those typically go in the stage dir instead of TEMPDIR, see other uses of this in the code base.

Reply:
Moved it to the stage dir.

Reply:
@haampie any suggestions?